src/lib.rs

```diff
@@ -187,6 +202,14 @@ mod tests {
     }
 
     #[test]
+    fn test_borrow_with() {
+        let lazycell = LazyCell::new();
+
+        let value = lazycell.borrow_with(|| 1);
+        assert_eq!(&1, value);
```

Comment:
Could you also test the positive case (i.e. `borrow_with(|| 1)` on a cell that's already been filled (just for complete test coverage).